### PR TITLE
🎄 Goodies

### PR DESCRIPTION
**Pull request by marcoscaceres**
🎄 New features:

   * 🎁  ReSpec now keeps a local copy of references and aliases in IDB. This means dramatically improved load times. 
   * 🎁  Support for `<a data-cite=spec#some-ref>`, so you can inline cite
   * 🎁  Support for `<a data-cite-frag=some-ref>,` which is the same above.

### CODE REVIEW

**Comment by marcoscaceres** [comment]
Ok, so Firefox 50 (the one served by travis is buggy, it seems). I can't reproduce locally (all tests pass) - but it passes in Chrome just fine, and same in Safari.

**Comment by marcoscaceres** [comment]
@halindrome, let me know how you want to handle review of this. I know it's a lot of code.

**Comment by halindrome** [comment]
I will try to dig into it soon. I've been on other work related stuff
lately!

**Comment by marcoscaceres** [comment]
I'm going to merge this in, as I need `data-cite` for the Web Payment Spec. Would still really appreciate a post merge review.

File: tests/spec/core/data-cite-spec.js [hunk 1]
```diff
@@ -65,4 +65,65 @@ describe("Core — data-cite attribute", () => {
...
+        const a = doc.querySelector("#t1 > a");
+        expect(a.textContent).toEqual("inline link");
+        expect(a.href).toEqual("https://html.spec.whatwg.org/multipage/#pass");
+        expect(a.hasAttribute("data-cite")).toEqual(false);
+        expect(doc.querySelector("#bib-WHATWG-HTML").closest("section").id)
+          .toEqual("normative-references");
+      }).then(done);
+    });
+  });
+});
```
Comment (halindrome) [inline tests/spec/core/data-cite-spec.js]: Newline missing?

**Comment by marcoscaceres** [comment]
Fyi, it's all transpiled to es5 so it works in older browsers :) it only gives the illusion of bleeding edge.

**Comment by marcoscaceres** [comment]
Sorry also about the size. This was a big one as it added a pretty important new feature (and it was basically my xmas break - I need to get a life;)). 

Anyway, I'll send an email to spec-prod explaining the new features. Some smaller bug fixes incoming.

**Pull request merged**